src/pool.rs

```diff
@@ -149,11 +230,37 @@ impl ConnectionPool {
         }
     }
 
-    // Connects to a MongoDB server as defined by the initial configuration.
-    fn connect(&self) -> Result<BufStream<TcpStream>> {
+
+    // Connects to a MongoDB server as defined by the initial
+    #[allow(unreachable_code)] // Suppresses warning for `panic` when ssl is enabled.
+    fn connect(&self) -> Result<BufStream<Stream>> {
         let host_name = &self.host.host_name;
         let port = self.host.port;
-        let stream = BufStream::new(try!(TcpStream::connect((&host_name[..], port))));
-        Ok(stream)
+        let inner_stream = try!(TcpStream::connect((&host_name[..], port)));
+
+        if let Some(SslConfig { ca_file: ref _ca_file,
+                                certificate_file: ref _certificate_file,
+                                key_file: ref _key_file }) = self.ssl {
+            #[cfg(feature = "ssl")]
+            {
+                let mut ssl_context = SslContext::builder(SslMethod::tls())?;
+                ssl_context.set_cipher_list("DEFAULT")?;
```

Comment:
Sorry about the wait! I've talked to some people, and apparently there's a TLS spec in the works. For now, I've gotten some detailed (but not finalized) information about how the drivers are supposed to configure TLS, so I'll set the options the way they're recommended when I update the PR.

There are a number of other things that we'll likely need to implement to be compliant with the spec down the line, so we'll likely have to update a few things once it's finalized. For now though, I think it's better to get a working implementation merged.